typing/ctype.ml

```diff
@@ -3523,12 +3556,16 @@ let matches env ty ty' =
   let snap = snapshot () in
   let vars = rigidify ty in
   cleanup_abbrev ();
-  let ok =
-    try unify env ty ty'; all_distinct_vars env vars
-    with Unify _ -> false
-  in
-  backtrack snap;
-  ok
+  try
```

Comment:
This would be clearer using `match ... with ... exception ...` instead of `try ... with ...`